source/puppet/5.5/lang_reserved.markdown

```diff
@@ -138,15 +132,15 @@ The names of classes and defined resource types can consist of one or more [name
 
 Namespace segments should match the following regular expression:
 
-    \A[a-z][a-z0-9_]*\Z
+    \A[a-z][a-zA-Z0-9_]*\Z
 
 The one exception is the top namespace, whose name is the empty string.
 
 Multiple namespace segments can be joined together in a class or defined type name with the `::` (double colon) [namespace][] separator.
 
 Class names with multiple namespaces should match the following regular expression:
 
-    \A([a-z][a-z0-9_]*)?(::[a-z][a-z0-9_]*)*\Z
+    \A([a-z][a-zA-Z0-9_]*)?(::[a-z][a-zA-Z0-9_]*)*\Z
```

Comment:
The actual regex used in the codebase is `\A((::)?[a-z]\w*)(::[a-z]\w*)*\z`. As far as I can tell, this is equivalent.

https://github.com/puppetlabs/puppet/blob/952414c36b45df9a0f46f4fe816bac4d2b18a89e/lib/puppet/pops/patterns.rb#L26

Reply:
Pushed a revision, and also emphasized that these are Ruby regex.